app/models/user.py

```diff
@@ -196,11 +197,16 @@ def _is_role(self, role_name, event_id=None):
         """
         Checks if a user has a particular Role at an Event.
         """
+        from app.models.users_groups_role import UsersGroupsRoles
+
         role = Role.query.filter_by(name=role_name).first()
         uer = UER.query.filter_by(user=self, role=role)
+        ugr = UsersGroupsRoles.query.filter_by(user=self, role=role, accepted=True)
         if event_id:
             uer = uer.filter_by(event_id=event_id)
-        return bool(uer.first())
+            event = Event.query.filter_by(id=event_id).first()
+            ugr = ugr.filter_by(group=event.group)
```

Comment:
event.group may be null

Test what happens in that case

Reply:
Then user event role will be tested.

Reply:
Will it crash or not or apply the permission regardless of group

Reply:
Then we should have check like this
if event.group: 
     event = Event.query.filter_by(id=event_id).first()
    ugr = ugr.filter_by(group=event.group)